storm-client/src/jvm/org/apache/storm/nimbus/ILeaderElector.java

```diff
@@ -52,6 +54,15 @@
      * @return the current leader's address , may return null if no one has the lock.
      */
     NimbusInfo getLeader();
+    
+    /**
+     * Wait for the caller to gain leadership. This should only be used in single-Nimbus clusters, and is only useful to allow testing
+     * code to wait for a LocalCluster's Nimbus to gain leadership before trying to submit topologies.
+     *
+     * @return true is leadership was acquired, false otherwise
+     */
+    @VisibleForTesting
+    boolean awaitLeadership(long timeout, TimeUnit timeUnit) throws InterruptedException;
```

Comment:
I assume we don't think ILeaderElector as user facing (public) API so don't mind about backward compatibility.

Reply:
No, I don't think this API is supposed to be used by anyone other than Nimbus.